File: 백준/Bronze/25501. 재귀의 귀재/재귀의 귀재.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n")
// let input = fs.readFileSync("./EX.txt").toString().trim().split('\n')

function PALINDROM(x, start, end, cnt) {
    cnt += 1
    if (start >= end) {
        return [1, cnt]
    } else if (x[start] !== x[end]) {
        return [0, cnt]
    } else {
        return PALINDROM(x, start+1, end-1, cnt)
    }
}

result = ""
for (let i = 1; i < input.length; i++) {
    s = input[i].trim()
    x = PALINDROM(s, 0, s.length-1, 0 )
    result += x.join(" ") + "\n"
}
console.log(result)

File: 백준/Bronze/25501. 재귀의 귀재/재귀의 귀재.py
import sys, math
input = sys.stdin.readline

def palindrom(s, start, end, cnt):
    cnt += 1
    if start >= end:
        return (1, cnt)
    elif s[start] != s[end]:
        return (0, cnt)
    else:
        return palindrom(s, start+1, end-1, cnt)

N = int(input())
for _ in range(N):
    S = input().rstrip()
    x = palindrom(S, 0, len(S)-1, 0)
    print(*x)
